packages/wonder-blocks-form/src/components/text-field.js

```diff
@@ -7,16 +7,48 @@ import Color from "@khanacademy/wonder-blocks-color";
 import Spacing from "@khanacademy/wonder-blocks-spacing";
 import {styles as typographyStyles} from "@khanacademy/wonder-blocks-typography";
 
-export default class TextField extends React.Component<{||}> {
+type State = {|
+    enabled: boolean,
+    disabled: boolean,
+    error: ?string,
+    focused: boolean,
+|};
+
+export default class TextField extends React.Component<{||}, State> {
+    state: State = {
+        enabled: true,
+        disabled: false,
+        error: null,
+        focused: false,
+    };
+
+    handleOnFocus: () => void = () => {
+        this.setState({
+            focused: true,
+        });
+    };
+
+    handleOnBlur: () => void = () => {
+        this.setState({
+            focused: false,
+        });
+    };
+
```

Comment:
As you are adding `state` and handlers here, this is a good point to start adding unit tests so we can verify that some behaviors in the component work as we intend :). Happy to chat more about this part in detail.